src/worker.js

```diff
@@ -1,8 +1,78 @@
-const worker = require('workerpool').worker
+const { worker } = require('workerpool')
 const { VM } = require('vm2')
 const { inspect } = require('util')
 
-const run = code =>
-  inspect(new VM().run(code), { depth: null, maxArrayLength: null })
+const run = async code => {
+  const vm = new VM({
+    sandbox: {
+      Set,
+      Map,
+      Date,
+      WeakSet,
+      WeakMap,
+      Buffer,
+      ArrayBuffer,
+      SharedArrayBuffer,
+      Int8Array,
+      Uint8Array,
+      Uint8ClampedArray,
+      Int16Array,
+      Uint16Array,
+      Int32Array,
+      Uint32Array,
+      Float32Array,
+      Float64Array,
+      BigInt64Array,
+      BigUint64Array,
+    },
+  })
+  const vmRegExpPrototype = vm.run('RegExp').prototype,
+    vmRegExpProtoToString = vmRegExpPrototype.toString
+  const primitiveTypes = [
+    'Number',
+    'String',
+    'Boolean',
+    'Symbol',
+    'BigInt',
+  ].map(type => {
+    const { prototype } = vm.run(type)
+    return [type, prototype, prototype.valueOf]
+  })
+  const promiseProtoThen = vm.run('Promise').prototype.then
+
+  let result = vm.run(code)
+  try {
+    result = (await promiseProtoThen.call(result, a => [a]))[0]
```

Comment:
```suggestion
    result = (await promiseProtoThen.call(result, value => value, reason => reason))
```

このように変更すると、`reject`された場合の値も送信することができました。

![image](https://user-images.githubusercontent.com/33122816/132994681-134d0719-97e8-4554-8853-d3a03094dd19.png)

ただ、余計なスタックトレースが表示されてて最悪になってしまった...